### PR TITLE
Add Deadline field

### LINKED ISSUE #64: Add Deadline Field

### PR DESCRIPTION
**Pull request by rahula1008**
Add Deadline field: d/dd-MM-yyyy 

Mandatory field
Input in dd-mm-yyyy format
Displays in MMM dd, yyyy format 

Closes #64

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -70,6 +74,7 @@ public JsonAdaptedPerson(Person source) {
         projectStatus = source.getProjectStatus().toString();
         paymentStatus = source.getPaymentStatus().toString();
         clientStatus = source.getClientStatus().toString();
+        deadline = source.getDeadline().value.format(Deadline.INPUT_FORMATTER);
```
Comment (jktang14) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:77]: I could be wrong here but I'm not sure if the format part is necessary considering the other parameters did not require such a checker

File: src/test/java/seedu/address/model/person/DeadlineTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void constructor_invalidDeadline_throwsIllegalArgumentException() {
+        String invalidDeadline = "";
+        assertThrows(IllegalArgumentException.class, () -> new Deadline(invalidDeadline));
+    }
+
+    @Test
+    public void isValidDeadline() {
```
Comment (jktang14) [inline src/test/java/seedu/address/model/person/DeadlineTest.java:23]: Comprehensive tests 👍

File: src/test/java/seedu/address/model/person/DeadlineTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        // null deadline
+        assertThrows(NullPointerException.class, () -> Deadline.isValidDeadline(null));
+
+        // invalid deadlines
+        assertFalse(Deadline.isValidDeadline("")); // empty string
+        assertFalse(Deadline.isValidDeadline(" ")); // spaces only
+        assertFalse(Deadline.isValidDeadline("31-02-2020")); // invalid date (February 31st doesn't exist)
+        assertFalse(Deadline.isValidDeadline("2020-12-12")); // incorrect format (should be dd-MM-yyyy)
+        assertFalse(Deadline.isValidDeadline("12/12/2020")); // slashes instead of dashes
+        assertFalse(Deadline.isValidDeadline("32-01-2020")); // invalid day
```
[the remark this answers is not in the document]
Comment (rahula1008) [reply src/test/java/seedu/address/model/person/DeadlineTest.java:33]: Good idea!

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 1 above)
Comment (jktang14) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:77]: how about just using `source.getDeadline().toString()`?

**Pull request merged**